File: transport/http/dto/attachment.go
package dto

import (
	"gav/internal/chat/model"

	"github.com/google/uuid"
)

type AttachmentRequest struct {
	URL      string               `json:"url"`
	Type     model.AttachmentType `json:"type"`
	FileName string               `json:"file_name"`
	FileSize int64                `json:"file_size"`
}

type AttachmentResponse struct {
	ID        uuid.UUID           	`json:"id"`
	MessageID uuid.UUID           	`json:"message_id"`
	URL       string              	`json:"url"`
	Type      model.AttachmentType 	`json:"type"`
	FileName  string              	`json:"file_name"`
	FileSize  int64               	`json:"file_size"`
}

File: Gav_App/Models/Attachment.swift
//
//  Attachment.swift
//  
//
//  Created by Виктория Кашуркина on 23.03.2026.
//


struct Attachment: Identifiable {
    let id: UUID
    let messageID: UUID
    let url: String
    let type: String
    let fileName: String
    let fileSize: Int64
}